Mage/Logger/Writers/ServerWriter.cs

```diff
@@ -1,13 +1,46 @@
-﻿using System.Collections.Generic;
+using System;
+using System.Collections.Generic;
+
+using Newtonsoft.Json.Linq;
 
 
 public class ServerWriter : LogWriter {
-	//private List<string> config;
+	protected Mage mage { get { return Mage.Instance; } }
+
+	private List<string> config;
+
 
 	public ServerWriter(List<string> logLevels) {
-		//config = logLevels;
+		config = logLevels;
+
+		Logger.logEmitter.on("log", HandleLog);
 	}
 
 	public override void Dispose() {
```

Comment:
This is kinda a general comment I think we should change the use of this, at the moment this will get called, then i think the GC will recall it as we are not supressing the GC for this class. (proper implementation of Dispose https://msdn.microsoft.com/en-us/library/fs2xkftw(v=vs.110).aspx)

We could also use destructors https://msdn.microsoft.com/en-us/library/66x5fx1b.aspx although I'm not so sure about using these.